api/src/main/java/marquez/common/models/SourceType.java

```diff
@@ -14,13 +14,42 @@
 
 package marquez.common.models;
 
-public enum SourceType {
-  BIGQUERY,
-  PULSAR,
-  ORACLE,
-  MYSQL,
-  POSTGRESQL,
-  REDSHIFT,
-  SNOWFLAKE,
-  KAFKA;
+import static marquez.common.base.MorePreconditions.checkNotBlank;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.util.StdConverter;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+@EqualsAndHashCode
+@ToString
+@JsonDeserialize(converter = SourceType.FromValue.class)
+@JsonSerialize(converter = SourceType.ToValue.class)
+public final class SourceType {
```

Comment:
What is the convention for using classes like this over conventional strings that are associated with typed objects?

Reply:
Mainly, it's an alternative to working with just the literal _string_ and provides the added context to method calls that can be self-documenting. For example:

```
// Using  literal strings
Source(String, String ,...)

// Using defined types
Source(SourceType, SourceName, ...)
```

For now, we have a simple check that the _value_ for a `SourceType` can not be _null_, or _blank_ but this can be expanded to check for allowed length, characters, etc (similar to [NamespaceName](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/common/models/NamespaceName.java)). You also get that extra assurance that you are working with a valid value when calling `SourceType.getValue()`.

Of course, not everything has to be a type (ex: the description field for most of our models) and it's more of a case-by-case basis.

Reply:
I'm fine with this pattern if we can move to a common object model and use visitor patterns to transform our objects between contexts. Currently its a bit onerous to have to always box and unbox String values to call functions.